CI.hs

```diff
@@ -30,15 +30,30 @@ main = do
               <> Opts.progDesc "Build ghc-lib and ghc-lib-parser tarballs."
               <> Opts.header "CI - CI script for ghc-lib"
           )
-  Options {ghcFlavor, noGhcCheckout, noBuilds, versionSuffix} <- Opts.execParser opts
-  version <- buildDists ghcFlavor noGhcCheckout noBuilds versionSuffix
+  Options {ghcFlavor, noGhcCheckout, noBuilds, versionSuffix, optsAllowNewer} <- Opts.execParser opts
+  let allowNewerSpecs = optsAllowNewer
+  let allowNewerFlags = mkAllowNewerFlags allowNewerSpecs
+  version <- buildDists ghcFlavor noGhcCheckout noBuilds versionSuffix allowNewerFlags allowNewerSpecs
   putStrLn version
 
+mkAllowNewerFlags :: [String] -> String
+mkAllowNewerFlags xs =
+  let render = unwords . map (\s -> "--allow-newer=" ++ s)
+      def = ""
+  in case xs of
+       [] -> if null def then "" else " " ++ def
+       _  -> " " ++ render xs
```

Comment:
Since def is now unused, we can reduce this to:

```suggestion
mkAllowNewerFlags [] = ""
mkAllowNewerFlags xs = " " ++ unwords (mkAllowNewerFlagsList xs)

mkAllowNewerFlagsList :: [String] -> [String]
mkAllowNewerFlagsList = map ("--allow-newer=" ++) xs
```

The `List` variant is introduced for later use in `ghc-lib-gen/src/Ghclibgen.hs`